deploy/salt/roots/salt/memex-explorer.sls

```diff
@@ -42,31 +39,38 @@ collectstatic:
     - require:
         - sls: conda-memex
 
-refresh_nginx:
+celery:
   cmd.run:
-    - name: |
-        /home/vagrant/miniconda/envs/memex/bin/python /vagrant/source/manage.py refresh_nginx
-    - cwd: /home/vagrant
+    - name: /home/vagrant/miniconda/envs/memex/bin/celery --detach --loglevel=debug --logfile=/vagrant/source/celeryd.log --workdir="/vagrant/source" -A memex worker
+    - cwd: /vagrant/source
     - user: vagrant
+    - env:
+        - JAVA_HOME: '/usr/lib/jvm/java-7-oracle'
+    - unless: "ps -p $(cat /vagrant/source/celeryd.pid)"
     - require:
         - sls: conda-memex
 
-create_apps:
+supervisor:
   cmd.run:
     - name: |
-        /home/vagrant/miniconda/envs/memex/bin/python /vagrant/source/manage.py create_apps_Tika_ES_Kibana
-    - cwd: /home/vagrant
+        source activate memex
+        supervisord -c /vagrant/deploy/supervisor.conf        
     - user: vagrant
+    - env:
+       - PATH:
+           /home/vagrant/miniconda/bin:/home/ubuntu/miniconda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
```

Comment:
See https://github.com/saltstack/salt/issues/21840 for a horrible way to avoid PATH hard-codes in Salt by appending to the PATH variable.  You could also just source /home/vagrant/miniconda/bin/memex if it's (like I thought) always in /home/vagrant.

Reply:
As a note, if/when https://github.com/saltstack/salt/pull/44300 gets merged, things will be much more elegant :)

Reply:
https://github.com/saltstack/salt/pull/44300's merged now, and will be in the next release of salt :)